### ISSUE TITLE
ctrl-c does not reliably interrupt binary data sent to stdout

### ISSUE THREAD

**Issue by cjfp** [mintty/mintty#1111]
cat /dev/random

It's very difficult to stop this with Ctrl-C.  The best way I've found is to close the window.  This happens in practice when I cat a binary file by mistake, or when I do diff -a between two binaries and the result is much longer than I expect.  I'm not sure if this is fixable or not.

mintty 3.5.0 (x86_64-pc-cygwin) [Windows 7601]

**Comment by BrianInglis** [comment]
With recent console handling changes, `Ctrl-O` `discard = ^O` etc. work again and issuing that followed by `Ctrl-C` allows me to terminate the command eventually. 
Another approach I have used is `Ctrl-Z` `susp = ^Z`:
```
[1]+  Stopped                 cat /dev/random
$ kill %1
[1]+  Terminated              cat /dev/random
```

**Comment by cjfp** [comment]
Thanks, Ctrl-O Ctrl-C works more reliably than Ctrl-C.  It's not 100% but it's better.  Success appears to be related to how much output there has been already.

I wasn't able to make Ctrl-Z work reliably, at least not like Ctrl-O Ctrl-C.

**Comment by mintty** [comment]
Catting binary stuff will confuse every terminal and make interrupt feel stalled. This is not specific to mintty.

**Comment by cjfp** [comment]
Yes, fair enough.  For reference: https://superuser.com/questions/637860/why-does-cat-dev-urandom-break-your-terminal

Maybe there's a way to force an interrupt via the right-click context menu?  Just an idea.  I'm happy enough with Ctrl-O Ctrl-C.

**Comment by mintty** [comment]
There is "Send Break" in the extended context menu (Control+right-click) which sends a BRK condition to the tty, which would by default raise an INT signal at the backend of the pty.
This is not so much more than ^C actually, as both can be reconfigured, caught or ignored.
It is also possible to get a KILL signal sent to the foreground process, using option UserCommands as described in the Tips wiki page.
